### PR TITLE
minor: port more window tests to sqllogictests

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

Part of https://github.com/apache/arrow-datafusion/issues/4495

Follow on to https://github.com/apache/arrow-datafusion/pull/5199

# Rationale for this change

I am trying to keep the testability of DataFusion reasonable. Sqllogictests are easier to add / update so we are trying to move stuff over there.

# What changes are included in this PR?

Port more window tests to window.slt

# Are these changes tested?
Yes all tests
# Are there any user-facing changes?

No

### CODE REVIEW

File: datafusion/core/tests/sqllogictests/test_files/window.slt [hunk 1]
```diff
@@ -398,3 +408,274 @@ WITH _sample_data AS (
...
+#//         "| -47613      | 3|",
+#//         "| -13474      | 3|",
+#//         "+----------------------------+-----------------+",
+#//     ];
+#//     assert_batches_eq!(expected, &actual);
+#//     Ok(())
+#// }
+
+# async fn window_frame_ranges_preceding_following_desc() -> Result<()> {
+query error DataFusion error: Internal error: Operator + is not implemented for types Int8(5) and Utf8("1"). This was likely caused by a bug in DataFusion's code and we would welcome that you file an bug report in our issue tracker
```
Comment (alamb) [inline datafusion/core/tests/sqllogictests/test_files/window.slt:530]: This one is strange to me -- it seems like a real bug -- if you agree I will file a ticket for it.

However, the corresponding query ran just fine in .rs rust tests 🤔

File: datafusion/expr/src/window_frame.rs [hunk 2]
```diff
@@ -164,7 +164,7 @@ pub fn regularize(mut frame: WindowFrame, order_bys: usize) -> Result<WindowFram
             }
         } else {
             return Err(DataFusionError::Plan(format!(
-                "With window frame of type RANGE, the ORDER BY expression must be of length 1, got {}", order_bys)));
+                "With window frame of type RANGE, the ORDER BY expression must be of length 1, got {order_bys}")));
```
Comment (alamb) [inline datafusion/expr/src/window_frame.rs:167]: clippy was complaining about this for me locally

**Comment by alamb** [comment]
cc @mustafasrepo  and @ozankabak

**Review by ozankabak — APPROVED** [review]
LGTM, we will use the SLT functionality for new tests too from now on. Some CI checks are failing though.

File: datafusion/core/tests/sqllogictests/test_files/window.slt (hunk 1 above)
Comment (alamb) [reply datafusion/core/tests/sqllogictests/test_files/window.slt:530]: filed https://github.com/apache/arrow-datafusion/issues/5346

**Review by jackwener — APPROVED** [review]
LGTM.

I prepare to merge this PR in one day in case of new conflict.

File: datafusion/core/tests/sqllogictests/test_files/window.slt (hunk 1 above)
Comment (jackwener) [reply datafusion/core/tests/sqllogictests/test_files/window.slt]: I think it's a bug, I also meet it in other code when disable `skip_failed`

**Comment by alamb** [comment]
Thanks everyone!

**Pull request merged**